doc: Enhance rule documentation (#50)
salesforce/eslint-plugin-lwc: docs/rules/valid-wire.md

# Validate `wire` decorator usage (valid-wire)

> The following restriction applies to `wire` decorator:

Reviewer: ```suggestion
The following restrictions apply to the `@wire` decorator:
```

> -   The `wire` decorator can only be applied to class fields and class methods.

Reviewer: ```suggestion
-   Apply the `wire` decorator to class fields and class methods only.
```

> -   The first argument of `wire` must be an identifier.

Reviewer: ```suggestion
-   The first argument of `@wire` must be an identifier.
```

> -   The second argument of `wire` is optional. If it is present, it must be an object literal.

Reviewer: ```suggestion
-   The second argument of `@wire` is optional. If it's present, it must be an object literal.
```
